python/kwiver/sprokit/adapters/embedded_pipeline.cxx

```diff
@@ -0,0 +1,154 @@
+/*ckwg +29
+ * Copyright 2020 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <sprokit/processes/adapters/embedded_pipeline.h>
+#include <sprokit/pipeline_util/load_pipe_exception.h>
+
+#include <python/kwiver/vital/util/pybind11.h>
+
+#include <memory>
+#include <fstream>
+
+using namespace pybind11;
+namespace kwiver {
+namespace sprokit {
+namespace python {
+
+// Publicist class to access protected methods
+class wrap_embedded_pipeline
+  : public embedded_pipeline
+{
+public:
+  using embedded_pipeline::connect_input_adapter;
+  using embedded_pipeline::connect_output_adapter;
+  using embedded_pipeline::update_config;
+};
+
+// Trampoline class to allow us to use virtual methods
+class embedded_pipeline_trampoline
+  : public embedded_pipeline
+{
+public:
+  using embedded_pipeline::embedded_pipeline;
+
+  bool connect_input_adapter() override;
+  bool connect_output_adapter() override;
+  void update_config(vital::config_block_sptr config) override;
+};
+
+void build_pipeline(embedded_pipeline& self, vital::path_t const& desc_file);
+
+}
+}
+}
+
+namespace ksp = kwiver::sprokit::python;
+
+PYBIND11_MODULE(embedded_pipeline, m)
+{
+  class_< kwiver::embedded_pipeline,
+        std::shared_ptr<kwiver::embedded_pipeline>,
+        ksp::embedded_pipeline_trampoline>(m, "EmbeddedPipeline")
```

Comment:
Again, a class-level docstring with an example would be very useful.

Reply:
Successful docstring added in 4c92416d1d4373d10a7e85ded265c3b5aef134fe